src/simple.rs

```diff
@@ -0,0 +1,66 @@
+use std::{
+    env::current_dir,
+    io::{stdin, stdout, Write},
+};
+
+use crate::core::MatchData;
+
+pub fn run(args: super::args::Args) {
+    let directory = args.path.unwrap_or_else(|| {
+        current_dir().expect("Cannot access current directory, run with -p <DIR> to select another")
+    });
+    let (sender, receiver) = std::sync::mpsc::channel::<MatchData>();
+
+    let collector = move || {
+        let mut collected = vec![];
+        while let Ok(data) = receiver.recv() {
+            println!("{}", data.path.display());
+            for reason in data.languages() {
```

Comment:
so is it language or reason?

Reply:
Well, I have changed it to `language` because it suits its type (`LangData`) better. The old name, `reasons`, is only used in the implementation and users of this crate, including your UI code, don't have to worry about it.

Reply:
so change `reason` to `language`